listenbrainz/webserver/static/js/src/CBReviewModal.tsx

```diff
@@ -0,0 +1,703 @@
+              onChange={this.handleInputChange}
+              required
+            />
+            <small>
+              &nbsp;You acknowledge and agree that your contributed reviews to
+              CritiqueBrainz are licensed under a Creative Commons
+              Attribution-ShareAlike 3.0 Unported or Creative Commons
+              Attribution-ShareAlike-NonCommercial license as per your choice
+              above. You agree to license your work under this license. You
+              represent and warrant that you own or control all rights in and to
+              the work, that nothing in the work infringes the rights of any
+              third-party, and that you have the permission to use and to
+              license the work under the selected Creative Commons license.
+              Finally, you give the MetaBrainz Foundation permission to license
+              this content for commercial use outside of Creative Commons
+              licenses in order to support the operations of the organization.
+            </small>
+          </label>
+        </div>
+      </div>
+    );
+  };
+
+  getModalFooter = (hasPermissions: boolean) => {
+    const { entityToReview, success } = this.state;
+
+    /* User hasn't logged into CB yet: prompt them to authenticate */
+    if (!hasPermissions)
+      return (
+        <a
+          href={`${window.location.origin}/profile/music-services/details/`}
+          className="btn btn-success"
+          role="button"
+        >
+          {" "}
+          Connect To CritiqueBrainz{" "}
+        </a>
+      );
+
+    /* Submit review button */
+    if (entityToReview && !success) {
+      const { reviewValidateAlert } = this.state;
+      const { critiquebrainzAuth } = this.context;
+
+      return (
+        <button
+          type="submit"
+          id="submitReviewButton"
+          className={`btn btn-success ${reviewValidateAlert ? "disabled" : ""}`}
+          onClick={() =>
+            this.submitReviewToCB(critiquebrainzAuth?.access_token)
+          }
+        >
+          Submit Review to CritiqueBrainz
+        </button>
+      );
+    }
+
+    /* default: close modal button */
+    return (
+      <button type="button" className="btn btn-default" data-dismiss="modal">
+        Close
+      </button>
+    );
+  };
+
+  render() {
+    const { loading } = this.state;
+    const { critiquebrainzAuth } = this.context;
+    const hasPermissions = this.hasPermissions(critiquebrainzAuth);
+
+    const modalBody = this.getModalBody(hasPermissions);
+    const modalFooter = this.getModalFooter(hasPermissions);
+
+    return (
+      <div
+        className="modal fade"
+        id="CBReviewModal"
+        tabIndex={-1}
+        role="dialog"
+        aria-labelledby="CBReviewModalLabel"
+        data-backdrop="static"
+      >
+        <div className="modal-dialog" role="document">
+          <form className="modal-content" onSubmit={(e) => e.preventDefault()}>
```

Comment:
Here you override the onSubmit of the form with a "submit" button with an onclick event pointing to this.submitReviewToCB(…).
With my suggestions above regarding that, I think you should be able to use
```suggestion
          <form className="modal-content" onSubmit={this.submitReviewToCB}>
```
and on your submit button remove the onClick handler.
The `type="submit"` you have on the button should trigger the method passed as `onSubmit` to the <form> element